src/napari_imagej/widgets/parameter_widgets.py

```diff
@@ -269,3 +270,33 @@ def choices(self):
     @choices.setter
     def choices(self, choices: ChoicesType):
         self.layer_select.choices = choices
+
+
+class SaveFileWidget(FileEdit):
+    """
+    A magicgui FileEdit that saves a new file
+    """
+
+    def __init__(self, **kwargs):
+        kwargs["mode"] = "w"
+        super().__init__(**kwargs)
+
+
+class OpenFileWidget(FileEdit):
+    """
+    A magicgui FileEdit that opens an existing file
+    """
+
+    def __init__(self, **kwargs):
+        kwargs["mode"] = "r"
+        super().__init__(**kwargs)
+
+
+class DirectoryWidget(FileEdit):
+    """
+    A magicgui FileEdit that opens a directory
+    """
+
+    def __init__(self, **kwargs):
+        kwargs["mode"] = "d"
+        super().__init__(**kwargs)
```

Comment:
Maybe add a disclaimer about why we're doing this.

Namely, that getting the kwargs we're setting here over to magicgui would be a much more involved change. And this gives us more control.